tests/Microsoft.Identity.Test.Unit/CoreTests/InstanceTests/GenericAuthorityTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Utils;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.CoreTests.InstanceTests
+{
+    [TestClass]
+    public class GenericAuthorityTests 
```

Comment:
@g7ed6e - still testing this, but hope we can get it in in the next release, in 1-2 weeks. 

Btw, it looks like the HttpDocumentRetriever actually makes 2 HTTP calls: 

<img width="562" alt="image" src="https://user-images.githubusercontent.com/12273384/220979479-8f9e5286-894e-461d-9daa-52cb0d53e785.png">

This makes sense for token validation, but for token acquisition we don't care about the public key of the identity provider.

Reply:
> @g7ed6e - still testing this, but hope we can get it in in the next release, in 1-2 weeks.

Thanks @bgavrilMS ! That's good news!